verification/src/transaction_verifier.rs

```diff
@@ -209,3 +225,161 @@ impl<'a> CapacityVerifier<'a> {
         Ok(())
     }
 }
+
+const LOCK_TYPE_FLAG: u64 = 1 << 63;
+const TIME_TYPE_FLAG: u64 = 1 << 62;
+const VALUE_MUSK: u64 = 0x00ff_ffff_ffff_ffff;
+
+/// RFC 0017
+#[derive(Copy, Clone, Debug)]
+struct ValidSince(u64);
```

Comment:
I prefer enum to struct:
```rust
enum ValidSince {
  None,
  AbsoluteNumber(u64),
  AbsoluteTimestamp(u64),
  RelativeNumber(u64),
  RelativeTimestamp(u64),
}
```
And impl From/To u64 trait, that will remove these helper methods and simplify code.

Reply:
tomayto tomato, I can't see big improvement, it just change the conditions check from function to pattern match,

Reply:
It's a common refactoring practice in other OOP language, it is called `Replace Conditional with Polymorphism`.

Reply:
Can't agree, If we support more bits flags, the enum will increment exponentially, there's no clear improvement.

Reply:
Do not have a conclusion, just use the current implementation for now.